lib/github.ml

```diff
@@ -182,15 +181,16 @@ let gh_link_of_string url_str =
   match Uri.host url with
   | None -> None
   | Some host ->
-  match String.chop_prefix path ~prefix:"/" with
+  match (if Stre.starts_with path "/" then Some (Stre.drop_prefix path "/") else None) with
+  (* match String.chop_prefix path ~prefix:"/" with *)
   | None -> None
   | Some path ->
-    let path = String.chop_suffix_if_exists ~suffix:"/" path |> flip Stre.nsplitc '/' |> List.map ~f:Web.urldecode in
+    let path = Stre.drop_suffix path "/" |> flip Stre.nsplitc '/' |> List.map Web.urldecode in
     let make_repo ~prefix ~owner ~name =
-      let base = String.concat ~sep:"/" (List.rev prefix) in
+      let base = String.concat "/" (List.rev prefix) in
       let scheme = Uri.scheme url in
       let html_base, api_base =
-        if String.is_suffix base ~suffix:"github.com" then gh_com_html_base owner name, gh_com_api_base owner name
+        if Stre.drop_suffix base "github.com" != base then gh_com_html_base owner name, gh_com_api_base owner name
```

Comment:
```suggestion
        if Stre.drop_suffix base "github.com" <> base then gh_com_html_base owner name, gh_com_api_base owner name
```

Reply:
Is there a difference between `!=` and `<>` if we're comparing non-mutables? Or is it a style/convention thing?

Reply:
```
# let s1 = "a" ^ "b";;
val s1 : string = "ab"
# let s2 = "a" ^ "b";;
val s2 : string = "ab"
# s1 = s2;;
- : bool = true
# s1 == s2;;
- : bool = false
# s1 <> s2;;
- : bool = false
# s1 != s2;;
- : bool = true
```